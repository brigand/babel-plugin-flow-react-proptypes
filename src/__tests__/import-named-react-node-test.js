const babel = require('babel-core');
const content = `
// @flow
import type { Node } from 'react';

type Props = {
  foo: Node,
}

const C = (props: Props) => <div />;
`;

it('import-named-react-node', () => {
  const res = babel.transform(content, {
    babelrc: false,
    presets: ['@babel/env', '@babel/react', '@babel/flow'],
    plugins: [
      '@babel/syntax-flow',
      require('../'),
      "@babel/plugin-proposal-class-properties"
    ],
  }).code;
  expect(res).toMatch(/_propTypes\[['"]default['"]]\.node\.isRequired/);
  expect(res).toMatchSnapshot();
});

it('import-named-react-node esm', () => {
  const res = babel.transform(content, {
    babelrc: false,
    presets: [['@babel/env', { modules: false }], '@babel/react', '@babel/flow'],
    plugins: [
      '@babel/syntax-flow',
      require('../'),
      "@babel/plugin-proposal-class-properties"
    ],
  }).code;
  expect(res).toMatch(/PropTypes\.node\.isRequired/);
  expect(res).toMatchSnapshot();
});
