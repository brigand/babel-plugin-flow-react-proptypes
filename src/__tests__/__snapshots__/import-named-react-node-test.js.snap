// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`import-named-react-node 1`] = `
"\\"use strict\\";

var _propTypes = _interopRequireDefault(require(\\"prop-types\\"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { \\"default\\": obj }; }

var C = function C(props) {
  return /*#__PURE__*/React.createElement(\\"div\\", null);
};

C.propTypes = {
  \\"foo\\": function foo() {
    return (typeof _propTypes[\\"default\\"].node === \\"function\\" ? _propTypes[\\"default\\"].node.isRequired ? _propTypes[\\"default\\"].node.isRequired : _propTypes[\\"default\\"].node : _propTypes[\\"default\\"].shape(_propTypes[\\"default\\"].node).isRequired).apply(this, arguments);
  }
};"
`;

exports[`import-named-react-node esm 1`] = `
"var C = function C(props) {
  return /*#__PURE__*/React.createElement(\\"div\\", null);
};

C.propTypes = {
  \\"foo\\": function foo() {
    return (typeof PropTypes.node === \\"function\\" ? PropTypes.node.isRequired ? PropTypes.node.isRequired : PropTypes.node : PropTypes.shape(PropTypes.node).isRequired).apply(this, arguments);
  }
};
import PropTypes from \\"prop-types\\";"
`;
